core/shared/src/main/scala-2.13+/fs2/Chunk213And3Compat.scala

```diff
@@ -21,9 +21,28 @@ private[fs2] trait Chunk213And3Compat[+O] { self: Chunk[O] =>
     }
     buf.result()
   }
+
+  /** Views this Chunk as a Scala immutable Seq.
+    * Contrary to all methods that start with _"to"_ (e.g. {{toVector}}, {{toArray}}),
+    * this method does not copy data.
+    */
+  def asSeq: Seq[O] =
+    new ChunkAsSeq(this)
+}
+
+private[fs2] final class ChunkAsSeq[+O](chunk: Chunk[O]) extends immutable.Seq[O] {
```

Comment:
Those three are the only abstract methods on `Seq` thus, the only ones we need to override.
However, I wonder if we should override other ones for performance's sake, things like `copyToArray` ring my bell, but not sure what you all think.

Reply:
Yep, I'm not very familiar with Scala collections API but `copyToArray` (and maybe `foreach`) seem useful.

Reply:
`knownSize` by default returns -1, but I guess that at least for a compacted `Chunk` it would be really helpful to return its real size.

Reply:
Just to clarify: would it make sense to inherit it from `immutable.IndexedSeq` instead of just `Seq`? I wonder because according to the docs, `Chunk` is
> Immutable, strict, finite sequence of values that supports _efficient index-based random access_ of elements

So to me it looks like the `IndexedSeq` matches the idea of "efficient index-based random access" pretty well.

Reply:
Not sure.

https://github.com/typelevel/fs2/blob/3c93811b7f8d367eb23b02456cd6b18dbaec9962/core/shared/src/main/scala/fs2/Stream.scala#L4468-L4470

Reply:
I see. But in that case the doc's header looks quite misleading. I feel it would be better to re-phrase it to emphasize that it may or may not support efficient random access, depending on the underlying collection it is build of.

Reply:
One more method to consider: `iterableFactory`. The point is that `Seq` is not a view-like class. Therefore, by default it tries to re-create itself in every single transformation or filtering method. E.g. `map`, `flatMap`, `filter`, etc. That means, in particular, that if `iterableFactory` is not overridden, then `map` called on such a collection will convert it into a default representation (which is `List` for `Seq`). Not sure if it is the behavior that users would expect. So perhaps it would be better to create a compacted version of `Chunk` in such cases or something like that.

Reply:
@satorg IMHO, the whole point of this method is to provide a copy-free view of `Chunk` as a `Seq`, whose main use case is precisely a `foreach` kind of interop.
Thus, while we do try to avoid pointless indirections for methods like `isEmpty` or `to*`, any real use of this collection will consume it and produce a real `Seq` _(like `List`)_.

We may be more explicit on the **Scaladoc** if you think that would be useful.

Reply:
However, what about using `ArraySeq` or `Vector` as the default `iterableFactory` which would be more akin to `Chunk` than `List`?
cc @satorg @armanbilge

Reply:
@satorg @armanbilge done, both the **Scaladoc** adjustment and the _override_ of `iterableFactory`.
Please, let me know what you both think.